hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1163,78 +1176,69 @@ public static HoodieData<HoodieRecord> convertMetadataToColumnStatsRecords(Hoodi
     }
   }
 
-  public static final String[] META_COLS_TO_ALWAYS_INDEX = {COMMIT_TIME_METADATA_FIELD, RECORD_KEY_METADATA_FIELD, PARTITION_PATH_METADATA_FIELD};
-  public static final Set<String> META_COL_SET_TO_INDEX = new HashSet<>(Arrays.asList(META_COLS_TO_ALWAYS_INDEX));
-
-  public static List<String> getColumnsToIndex(HoodieTableConfig tableConfig,
-                                               HoodieMetadataConfig metadataConfig,
-                                               List<String> columnNames,
-                                               boolean overrideEnableCheck) {
-    return getColumnsToIndex(tableConfig, metadataConfig, Either.left(columnNames), overrideEnableCheck);
+  @VisibleForTesting
+  static final String[] META_COLS_TO_ALWAYS_INDEX = {COMMIT_TIME_METADATA_FIELD, RECORD_KEY_METADATA_FIELD, PARTITION_PATH_METADATA_FIELD};
+  private static final Set<String> META_COL_SET_TO_INDEX = new HashSet<>(Arrays.asList(META_COLS_TO_ALWAYS_INDEX));
 
-  }
-  
   public static List<String> getColumnsToIndex(HoodieTableConfig tableConfig,
                                                HoodieMetadataConfig metadataConfig,
                                                List<String> columnNames) {
-    return getColumnsToIndex(tableConfig, metadataConfig, columnNames, false);
+    return getColumnsToIndex(tableConfig, metadataConfig, Either.left(columnNames), Option.empty());
```

Comment:
We could get the recordMerger from HoodieWriteConfig and fetch the record type right?
why set it to Option.empty ?

Reply:
Punted this for now, since this requires changing the core MDT writer apis to take in recordMerger as an argument.